packages/thirdweb/src/react/web/ui/prebuilt/NFT/NFT.tsx

```diff
@@ -0,0 +1,266 @@
+ * you must wrapped it with React.Suspense to make it work.
+ *
+ * @component
+ * @example
+ * ### Basic usage
+ * ```tsx
+ * import { getContract } from "thirdweb";
+ * import { NFT } from "thirdweb/react";
+ *
+ * const nftContract = getContract({
+ *   address: "0x...",
+ *   chain: ethereum,
+ *   client: yourThirdwebClient,
+ * });
+ *
+ * <NFT contract={nftContract} tokenId={0n}>
+ *   This will show the media for tokenId #0 from the `nftContract` collection
+ *   <Suspense fallback={"Loading media..."}>
+ *     <NFT.Media />
+ *   </Suspense>
+ * </NFT>
+ * ```
+ */
+NFT.Media = (props: NFTMediaProps) => {
+  const { contract, tokenId } = useNFTContext();
+  const nftQuery = useSuspenseQuery({
+    queryKey: [
+      "__nft_info_internal__",
+      contract.chain.id,
+      contract.address,
+      tokenId.toString(),
+    ],
+    queryFn: () => getNFTInfo({ contract, tokenId }),
+  });
+  const animation_url = nftQuery.data?.metadata.animation_url;
+  const image =
+    nftQuery.data?.metadata.image || nftQuery.data?.metadata.image_url;
+
+  return (
+    <MediaRenderer
+      client={contract.client}
+      src={animation_url || image}
+      poster={image}
+      {...props}
+    />
+  );
+};
+
+/**
+ * This component fetches and displays an NFT's name. It takes in a `className` and `style` props
+ * so you can style it just like how you would style a <div> element.
+ * @returns A <div> element
+ *
+ * Since this component has an internal loading state (for when the NFT media is being fetched),
+ * you must wrapped it with React.Suspense to make it work.
+ *
+ * @component
+ * @example
+ * ### Basic usage
+ * ```tsx
+ * import { getContract } from "thirdweb";
+ * import { NFT } from "thirdweb/react";
+ *
+ * const nftContract = getContract({
+ *   address: "0x...",
+ *   chain: ethereum,
+ *   client: yourThirdwebClient,
+ * });
+ *
+ * <NFT contract={nftContract} tokenId={0n}>
+ *   This will show the name for tokenId #0 from the `nftContract` collection
+ *   <Suspense fallback={"Loading media..."}>
+ *     <NFT.Name className="mx-auto" style={{ color: "red" }} />
+ *   </Suspense>
+ * </NFT>
+ * ```
+ */
+NFT.Name = (props: { className?: string; style?: React.CSSProperties }) => {
+  const { contract, tokenId } = useNFTContext();
+  const nftQuery = useSuspenseQuery({
+    queryKey: [
+      "__nft_info_internal__",
+      contract.chain.id,
+      contract.address,
+      tokenId.toString(),
+    ],
+    queryFn: () => getNFTInfo({ contract, tokenId }),
+  });
+  const name = nftQuery.data?.metadata.name || "";
+  return <div {...props}>{name}</div>;
+};
+
+/**
+ * This component fetches and displays an NFT's description. It takes in a `className` and `style` props
+ * so you can style it just like how you would style a <div> element.
+ * @returns A <div> element
+ *
+ * Since this component has an internal loading state (for when the NFT media is being fetched),
```

Comment:
Does it really *require* suspense? Or will it just not show the component until the query finishes if we don't have suspense. Maybe "recommended" is what it is